### PR TITLE
Relative message numbers

### PR DESCRIPTION
**Pull request by flatcap**
This contains:
- bug fix for a NULL context
- bug fix for the old GNU regex
- some light tidying

/cc @nobrowser, @gahr

### CODE REVIEW

**Comment by flatcap** [comment]
@nobrowser: @guckes noticed that the patterns don't seem to work with `<limit>`
Please could you have a look.

**Comment by nobrowser** [comment]
@flatcap , @guckes : can you be more specific about <limit> ? What is the scenario, what happens, and how is it different from what you expected?  Also, is it a regression from the existing code?

File: pattern.c [hunk 1]
```diff
@@ -509,6 +509,19 @@ is_context_available(BUFFER *s, regmatch_t pmatch[], int kind, BUFFER *err)
...
+#define RANGE_NUM_RX "([0-9]+|0x[0-9a-fA-F]+)[MmKk]?"
+#define RANGE_REL_SLOT_RX "[ \t]*([.^$]|-?" RANGE_NUM_RX ")?[ \t]*"
+#define RANGE_REL_RX "^" RANGE_REL_SLOT_RX "," RANGE_REL_SLOT_RX
+#define RANGE_ABS_SLOT_RX "[ \t]*([.^$]|" RANGE_NUM_RX ")?[ \t]*"
+#define RANGE_ABS_RX "^" RANGE_ABS_SLOT_RX "-" RANGE_ABS_SLOT_RX
+#define RANGE_LT_RX "^()[ \t]*(<[ \t]*" RANGE_NUM_RX ")[ \t]*"
+#define RANGE_GT_RX "^()[ \t]*(>[ \t]*" RANGE_NUM_RX ")[ \t]*"
+#define RANGE_BARE_RX "^[ \t]*([.^$]|" RANGE_NUM_RX ")[ \t]*"
+#else
+/* The regexes in a modern format */
```
Comment (nobrowser) [inline pattern.c]: Thanks :)

**Comment by flatcap** [comment]
I've just given `<limit>` a bit of a test and it seems fine to me.

**Comment by toogley** [comment]
(not sure if @guckes criticism is solved or not)

**Comment by nobrowser** [comment]
@toogley, I don't know what the criticism is.  Do you have a link?

**Comment by toogley** [comment]
@nobrowser nope, i was just referring to [flatcap's comment](https://github.com/neomutt/neomutt/pull/311#issuecomment-274260773) - i guess they talked in IRC or so about that. And Sven hasn't responded to that (but i don't know why, i thought just it wasn't noticed - therefore my change to the `status:in-progress` label)

**Comment by nobrowser** [comment]
I (and apparently @flatcap as well) tried limiting with all the new syntaxes and it does work.

Of course it's possible @guckes meant some more complex scenario, that's why I was asking ...

**Pull request closed without merging**